src/fireedge/src/server/routes/api/oneflow/utils.js

```diff
@@ -0,0 +1,99 @@
+ * Copyright 2002-2025, OpenNebula Project, OpenNebula Systems               *
+ *                                                                           *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may   *
+ * not use this file except in compliance with the License. You may obtain   *
+ * a copy of the License at                                                  *
+ *                                                                           *
+ * http://www.apache.org/licenses/LICENSE-2.0                                *
+ *                                                                           *
+ * Unless required by applicable law or agreed to in writing, software       *
+ * distributed under the License is distributed on an "AS IS" BASIS,         *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  *
+ * See the License for the specific language governing permissions and       *
+ * limitations under the License.                                            *
+ * ------------------------------------------------------------------------- */
+
+const { request: axios } = require('axios')
+const btoa = require('btoa')
+
+const { defaultOneFlowServer } = require('server/utils/constants/defaults')
+const { getFireedgeConfig } = require('server/utils/yml')
+const { httpMethod } = require('server/utils/constants/defaults')
+const { addPrintf } = require('server/utils/general')
+
+const { GET, DELETE } = httpMethod
+
+const appConfig = getFireedgeConfig()
+
+/**
+ * Return schema error.
+ *
+ * @param {string} error - get error for schema
+ * @returns {string} schema
+ */
+const returnSchemaError = (error = []) =>
+  error
+    .map((element) => (element && element.stack ? element.stack : ''))
+    .toString()
+
+/**
+ * Parse number to int.
+ *
+ * @param {string} validate - number to validate
+ * @returns {number}number to int
+ */
+const parseToNumber = (validate) =>
+  isNaN(parseInt(validate, 10)) ? validate : parseInt(validate, 10)
+
+/**
+ * Connection to one flow server.
+ *
+ * @param {object} requestData - data for request
+ * @param {Function} success - callback success function
+ * @param {Function} error - callback error function
+ */
+const oneFlowConnection = (
+  requestData = {},
+  success = () => undefined,
+  error = () => undefined
+) => {
+  const { method, path, user, password, request, post } = requestData
+  const optionMethod = method || GET
+  const optionPath = path || '/'
+  const optionAuth = btoa(`${user || ''}:${password || ''}`)
+
+  const options = {
+    method: optionMethod,
+    baseURL: appConfig.oneflow_server || defaultOneFlowServer,
+    url: request ? addPrintf(optionPath, request || '') : optionPath,
+    headers: {
+      Authorization: `Basic ${optionAuth}`,
+    },
+    validateStatus: (status) => status >= 200 && status < 400,
+  }
+
+  if (post) options.data = post
+
+  axios(options)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Server-Side Request Forgery (SSRF) vulnerability detected.  Untrusted user input is being used in a network request (in fetch, axios, or http.request)  without proper validation. This can allow attackers to craft requests to internal systems, access metadata services,  or perform internal port scans. To fix this issue, strictly validate and sanitize the input URL  using a library like `node:url` for parsing, enforce an allowlist of trusted domains,  and block access to private IP ranges (e.g., 127.0.0.1, 169.254.169.254).  Learn more: https://owasp.org/www-community/attacks/Server_Side_Request_Forgery